packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -1355,25 +1298,70 @@ export default abstract class SqlIntegration
     return `LEAST(${cap}, ${value})`;
   }
 
-  private getAggregateMetricColumn(metric: MetricInterface, alias = "m") {
+  private addPrePostTimeFilter(
+    col: string,
+    timePeriod: MetricAggregationType
+  ): string {
+    const mcol = `m.timestamp`;
+    if (timePeriod === "pre") {
+      return `${this.ifElse(`${mcol} < d.preexposure_end`, `${col}`, `NULL`)}`;
+    }
+    if (timePeriod === "post") {
+      return `${this.ifElse(
+        `${mcol} >= d.conversion_start`,
```

Comment:
Rather than checking if `m.user_id` or `m.timestamp` is NULL, I'm just going to check that the metric is after the start of the conversion window, which will return NULL if `m.timestamp` is missing which will effectively be the same. This allows us to use the same code for the post period for RA and the regular non-RA metric aggregation.

It also should be just as performant as `m.user_id IS NOT NULL` because it is still just one comparison.